src/Paprika/Data/SlottedArray.cs

```diff
@@ -177,25 +175,25 @@ public bool MoveNext()
             if (index < to)
             {
                 _index = index;
-                _current = Build();
+                Build(out _current);
                 return true;
             }
 
             return false;
         }
 
-        public Item Current => _current;
+        public readonly Item Current => _current;
 
-        private Item Build()
+        private void Build(out Item value)
         {
             ref var slot = ref _map[_index];
             var span = _map.GetSlotPayload(ref slot);
-            var key = Slot.UnPrepareKey(span, slot.Hash, slot.KeyPreamble, _bytes.Span, out var data);
+            var key = Slot.UnPrepareKey(slot.Hash, slot.KeyPreamble, span, _bytes.Span, out var data);
```

Comment:
What is the reason for the reorg? Should not big param go first?

Reply:
https://learn.microsoft.com/en-us/cpp/build/x64-calling-convention?view=msvc-170#calling-convention-defaults

Only 4 parameters are passed by register; for instance that's reduced by 1, for generic reduced by 1 more; anything larger than 8 bytes goes via stack any thing beyond 4 params goes via stack.

<img width="470" alt="image" src="https://github.com/NethermindEth/Paprika/assets/1142958/558b7894-bd7e-410a-8ac0-a70847601c7f">


<img width="473" alt="image" src="https://github.com/NethermindEth/Paprika/assets/1142958/2289c82a-b0e5-4f0e-8f4a-fbb55a95a6d4">

So you want smaller parameters first (including `in` and `ref`) since any large parameters will need to go on stack anyway, so why waste a register position?

Reply:
This is clearly not how remembered it. Thank you.